apps/codeforafrica/src/components/Projects/Projects.js

```diff
@@ -57,10 +55,14 @@ const Projects = React.forwardRef(function Projects(
   }, [data]);
 
   useEffect(() => {
-    router.push(queryParams, undefined, {
-      scroll: true,
-      shallow: true,
-    });
+    if (router.isReady) {
+      const [pathname] = router.asPath.split("?");
+      const url = pathname ? `${pathname}${queryParams}` : queryParams;
+      router.push(url, undefined, {
```

Comment:
Just for my understanding, if `url` we're passing to `router.push` doesn't start with `/`, isn't the `router` smart enough to keep us on the same page and just update the query part? For example, if my current page is `/projects/pesayetu` and I run:
```js
    router.push('?page=2', undefined, { scroll: true, shallow: true });
```
Isn't my new address going to be `/projects/pesayetu?page=2`?

Basically, do I always need to include `/projects/pesayetu` even if I'm just changing the query part?

Reply:
@kilemensi no, the new address will look something like `/[...slugs]?page=2`

Reply:
Interesting @koechkevin ... then if `queryParams` where an object, using the object form of [`push`](https://nextjs.org/docs/pages/api-reference/functions/use-router#with-url-object) would have worked best here but that ship has sailed.

Rather than doing the whole split thing, `window.location.pathname` should provide us with what we need, no?

Reply:
That also works but we will have to disable eslint ` // eslint-disable-next-line no-undef` my preference would still be asPath and check if `router.isReady`

Reply:
We don't need to disable eslint... you just need to tell eslint this file is running on a [browser](https://github.com/CodeForAfrica/ui/blob/40ec020db553f86d37fc2c992f0f05407574c751/apps/codeforafrica/src/pages/admin/index.page.js#L1).

>my preference would still be asPath and check if router.isReady
Why do that if browser already gives you the value?